File: practice/algorithms/implementation/service_lane/ServiceLane.scala
object ServiceLane extends App {
  val lines = io.Source.stdin.getLines().toList
  val lanes = lines(1).split(" ").map(_.toInt)
  val segments = lines.drop(2)
  val vehicles = segments.map(width(lanes, _))
  println(vehicles.mkString("\n"))

  def width(lanes: Array[Int], inputLane: String): Int = {
    val arr = inputLane.split(" ").map(_.toInt)
    lanes.slice(arr(0), arr(1) + 1).min
  }
}

File: practice/algorithms/implementation/service_lane/service_lane.rb
(length, tests) = gets.split(" ").map{ |i| i.to_i }
lanes = gets.split(" ").map{ |i| i.to_i }

tests.times do
  arr = gets.split(" ").map{ |i| i.to_i }
  puts lanes.slice(arr[0], arr[1] - arr[0] + 1).min
end